spec/milestone_checker_spec.rb

```diff
@@ -0,0 +1,151 @@
+# frozen_string_literal: true
+
+require_relative 'spec_helper'
+
+module Danger
+  describe Danger::MilestoneChecker do
+    it 'is a plugin' do
+      expect(described_class.new(nil)).to be_a Danger::Plugin
+    end
+
+    describe 'with Dangerfile' do
+      before do
+        @dangerfile = testing_dangerfile
+        @plugin = @dangerfile.milestone_checker
+      end
+
+      it 'reports a warning when a PR has a milestone with due date within the warning days threshold' do
+        pr_json = {
+          'milestone' => {
+            'title' => 'Release Day',
+            'html_url' => 'https://wp.com',
+            'due_on' => '2023-06-30T23:59:01Z'
+          },
+          'state' => 'open'
+        }
+
+        date_one_day_before_due = DateTime.parse('2023-06-29T23:59:01Z')
+
+        allow(@plugin.github).to receive(:pr_json).and_return(pr_json)
+        allow(DateTime).to receive(:now).and_return(date_one_day_before_due)
+
+        @plugin.check_milestone_due_date
+
+        expected_warning = ["This PR is assigned to the milestone [Release Day](https://wp.com) which is expiring in less than 5 days.\nPlease make sure to get it merged by then or assign it to a later expiring milestone."]
```

Comment:
Nitpick

```suggestion
        expected_warning = ["This PR is assigned to the milestone [Release Day](https://wp.com) which is expiring in less than 5 days.\nPlease make sure to get it merged by then or assign it to a later milestone."]
```

```suggestion
        expected_warning = ["This PR is assigned to the milestone [Release Day](https://wp.com) which is expiring in less than 5 days.\nPlease make sure to get it merged by then or assign it to a milestone expiring later."]
```

I also wonder if "expiring" is the right verb to use here 🤔 . What to you think of "due to finish" or "with deadline"?

Reply:
I had the same question with expiring / closing or just say its due date already passed and related. I think "expiring" is not necessarily wrong but it sounds maybe too informal. I'll try rewording it.